File: PHP/src/Controller/CategorieController.php
<?php

namespace App\Controller;

use App\Config\AbstractController;
use App\Entity\Categorie;
use App\Services\CategorieService;

class CategorieController extends AbstractController
{
    private CategorieService $categorieService;

    public function __construct()
    {
        $this->categorieService = new CategorieService();
    }

    public function index()
    {
        $utilisateur = $_SESSION["user"];
        $categories = $this->categorieService->lister($utilisateur->getIdUtilisateur());
        $this->render("categorie/list",compact("categories"));
    }

    public function create()
    {
        if ($_SERVER["REQUEST_METHOD"] === "POST") {
            $categorie = new Categorie();
            $categorie->setNom($_POST["nom"]);
            $categorie->setType($_POST["type"]);
            $categorie->setParDefaut(false);
            $categorie->setIdUtilisateur($_SESSION["user"]->getIdUtilisateur());
            $this->categorieService->ajouter($categorie);
            $this->redirect("?page=categorie");
        }
        $this->render("categorie/form");
    }

    public function edit()
    {
        $categorie = $this->categorieService->rechercherParId($_GET["id"]);
        if ($_SERVER["REQUEST_METHOD"] === "POST") {
            $categorie->setNom($_POST["nom"]);
            $categorie->setType($_POST["type"]);
            $this->categorieService->modifier($categorie);
            $this->redirect("?page=categorie");
        }
        $this->render("categorie/form",compact("categorie"));
    }

    public function delete()
    {
        $this->categorieService->supprimer($_GET["id"]);
        $this->redirect("?page=categorie");
    }
}

File: JAVA/final-BT-java/src/controllers/CategorieController.java
package controllers;

import models.Categorie;
import services.CategorieService;

import java.util.List;

public class CategorieController {
    private final CategorieService categorieService;
    public CategorieController() {
        categorieService = new CategorieService();
    }
    public boolean ajouter(Categorie categorie) {
        return categorieService.ajouter(categorie);
    }
    public boolean modifier(Categorie categorie) {
        return categorieService.modifier(categorie);
    }
    public boolean supprimer(int idCategorie) {
        return categorieService.supprimer(idCategorie);
    }
    public List<Categorie> listerUtilisateur() {
        return categorieService.listerUtilisateur();
    }
    public List<Categorie> listerCommunes() {
        return categorieService.listerCommunes();
    }
}